src/wallet/rpcdump.cpp

```diff
@@ -725,14 +728,38 @@ UniValue z_importviewingkey(const UniValue& params, bool fHelp)
     if (!IsValidViewingKey(viewingkey)) {
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid viewing key");
     }
-    // TODO: Add Sapling support. For now, return an error to the user.
-    if (boost::get<libzcash::SproutViewingKey>(&viewingkey) == nullptr) {
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Currently, only Sprout viewing keys are supported");
-    }
-    auto vkey = boost::get<libzcash::SproutViewingKey>(viewingkey);
-    auto addr = vkey.address();
 
-    {
+    if (boost::get<libzcash::SaplingIncomingViewingKey>(&viewingkey) == nullptr) {
```

Comment:
This logic seems wrong, shouldn't it be `!=`? At least, importing worked for me by replacing `==` with `!=` here and below when comparing for a Sprout viewing key

Reply:
I confirmed that running the rpc test fails with the `==` above, but changing them to `!=` fixed it

Reply:
That's true, I don't know how I missed that, fixed.